webapp/lib/view.dart

```diff
@@ -192,3 +179,135 @@ class AvailablePackagesViewPartial {
+        ..text = tag;
+      tagItem.onClick.listen((event) {
+        var selectedTag = (event.target as Element);
+        controller.command(controller.UIAction.configurationTagSelected, new controller.ConfigurationTagData(selectedTag: selectedTag.text.trim()));
+        tagListElement.children.forEach((t) => t.classes.remove('configure-package__tag-item-active'));
+        selectedTag.classes.add('configure-package__tag-item-active');
+      });
+      tagListElement.append(tagItem);
+    });
+    tagListElement.children.first.classes.add('configure-package__tag-item-active');
+    tagListElement.append(
+      new ButtonElement()
+        ..classes.add('configure-package__button-add-tag-action')
+        ..text = '+'
+        ..onClick.listen((event) => tagListElement.append(addTagDropDown(controller.addtitionalTagData.keys.toList())))
+    );
+  }
+
+  DivElement addTagDropDown(List<String> tags) {
+    var addTagModal = new DivElement()
+      ..classes.add('configure-package__response-add-tag-modal');
+    addTagModal.append(
+      HeadingElement.h6()
+      ..classes.add('configure-package__response-add-tag-modal-heading')
+      ..text = 'Select new tag to add');
+    var tagOptions = new SelectElement()
+      ..classes.add('configure-package__response-add-tag-modal-dropdown')
+      ..onChange.listen((event) {
+        var selectedOption = (event.currentTarget as SelectElement).value;
+        controller.command(controller.UIAction.addConfigurationTag, new controller.ConfigurationTagData(tagToAdd: selectedOption));
+        if(tagListElement.children.last is DivElement) tagListElement.children.removeLast();
+      });
+    tagOptions.add(new OptionElement()..text = '--Select Tag--', false);
+    tags.forEach((tag) {
+      var option = new OptionElement()
+        ..text = tag
+        ..value = tag;
+      tagOptions.add(option, false);
+    });
+    addTagModal.append(tagOptions);
+    return addTagModal;
+  }
+}
+
+class ConfigurationViewTagResponsesPartial {
+  DivElement tagResponsesElement;
+  DivElement _tagResponsesHeader;
+  DivElement _tagResponsesBody;
+
+  ConfigurationViewTagResponsesPartial() {
+    tagResponsesElement = new DivElement()
+      ..classes.add('configure-package__tag-responses');
+    _tagResponsesHeader = new DivElement()
+      ..classes.add('configure-package__tag-responses-header');
+    _tagResponsesBody = new DivElement()
+      ..classes.add('configure-package__tag-responses-body');
+  }
+
+  void renderResponses(String tag, Map<String, List<String>> responses) {
+    tagResponsesElement.children.clear();
+    _tagResponsesHeader.children.clear();
+    _tagResponsesBody.children.clear();
+    responses.forEach((language, responseSet) {
+      _tagResponsesHeader.append(new HeadingElement.h5()..text = language);
+      var body = new DivElement()..classes.add('configure-package__tag-responses-body-items');
+      int responseIndex = 0;
+      for (var response in responseSet) {
```

Comment:
if you need to iterate through a list and keep track of an index, prefer to use a `for i` loop, e.g.
```suggestion
      for (int i = 0; i < responseSet.length; i++) {
        var response = responseSet[i];
```

Reply:
Yes. This is the right one.